test/test_field_reductions.jl

```diff
@@ -235,6 +235,18 @@ trilinear(x, y, z) = x + y + z
 
             set!(c, (x, y) -> y)
             @test maximum(c) == CUDA.@allowscalar grid.yᵃᶜᵃ[3]
+
+            underlying_grid = RectilinearGrid(arch, size = (1, 1, 8), extent=(1, 1, 1))
+
+            grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom((x, y) -> -3/4))
+            c = Field((Center, Center, Center), grid)
+
+            set!(c, (x, y, z) -> -z)
+            @test maximum(c) == CUDA.@allowscalar c[1, 1, 3]
+            @test CUDA.@allowscalar compute!(Field(Average(c, condition=(c .< 0.5))))[1, 1, 1] == 0.25
+            zᶜᶜᶜ = KernelFunctionOperation{Center, Center, Center}(znode, grid, Center(), Center(), Center())
+            avg_bottom_half_not_immersed = CUDA.@allowscalar (c[1, 1, 3] + c[1, 1, 4]) / 2
```

Comment:
Can this be written without `allowscalar`?

Reply:
I removed all new `@allowscalar` statements along with a couple more that were already there before and weren't needed.

I'm now converting the CuArrays to Arrays, but I'm not sure if that's the best way to go. The other alternative is hardcoding the results, but then the whether the tests pass or not would depend on grid size, which is why I avoided that.